lib/mp4/transmuxer.js

```diff
@@ -119,7 +134,66 @@ AudioSegmentStream = function(track) {
     this.trigger('data', {track: track, boxes: boxes});
     this.trigger('done');
   };
+
+  // If the audio segment extends before the earliest allowed dts
+  // value, remove AAC frames until starts at or after the earliest
+  // allowed DTS
+  this.trimAacFramesByEarliestDts_ = function(aacFrames) {
+    if (track.minSegmentDts >= earliestAllowedDts) {
+      return aacFrames;
+    }
+
+    // We will need to recalculate the earliest segment Dts
+    track.minSegmentDts = Infinity;
+
+    return aacFrames.filter(function(currentFrame) {
+      // If this is an allowed frame, keep it and record it's Dts
+      if (currentFrame.dts >= earliestAllowedDts) {
+        track.minSegmentDts = Math.min(track.minSegmentDts, currentFrame.dts);
+        track.minSegmentPts = track.minSegmentDts;
+        return true;
+      }
+      // Otherwise, discard it
+      return false;
+    });
+  };
+
+  // generate the track's raw mdat data from an array of frames
+  this.generateSampleTable_ = function(frames) {
+    var
+      i,
+      currentFrame,
+      samples = [];
+
+    for (i = 0; i < frames.length; i++) {
+      currentFrame = frames[i];
+
+      samples.push({
+        size: currentFrame.data.byteLength,
+        duration: 1024 // For AAC audio, all samples contain 1024 samples
+      });
+    }
+    return samples;
+  };
+
+  // generate the track's sample table from an array of frames
+  this.concatenateFrameData_ = function(frames) {
+    var
+      i,
+      currentFrame,
+      dataOffset = 0,
+      data = new Uint8Array(sumByteLengths(frames, 'data'));
```

Comment:
We aren't doing byte-wise iteration at all. Instead simply iterating over each *frame* to sum the lengths. It's exactly the same work as before just in it's own function. Also this is audio which is an order of magnitude less data (at least) than the video side.